src/test/java/tdl/s3/sync/progress/UploadStatsProgressListenerTest.java

```diff
@@ -0,0 +1,54 @@
+package tdl.s3.sync.progress;
+
+import java.io.File;
+import org.junit.Test;
+import static org.junit.Assert.*;
+import org.junit.Before;
+import static org.mockito.Mockito.*;
+
+public class UploadStatsProgressListenerTest {
+
+    private UploadStatsProgressListener listener;
```

Comment:
It would be nice to use an Acceptance test for this functionality instead of a unit test.
The integration with the Sync is quite important and we rely on it for `record-and-upload`